packages/block-editor/src/components/block-list/block.js

```diff
@@ -372,34 +370,24 @@ function BlockListBlock( {
 				event.preventDefault();
 			}
 
-		// Avoid triggering multi-selection if we click toolbars/inspectors
-		// and all elements that are outside the Block Edit DOM tree.
-		} else if ( blockNodeRef.current.contains( event.target ) ) {
-			isPointerDown.current = true;
-
-			// Allow user to escape out of a multi-selection to a singular
-			// selection of a block via click. This is handled here since
-			// onFocus excludes blocks involved in a multiselection, as
-			// focus can be incurred by starting a multiselection (focus
-			// moved to first block's multi-controls).
-			if ( isPartOfMultiSelection ) {
-				onSelect();
-			}
+		// Allow user to escape out of a multi-selection to a singular
+		// selection of a block via click. This is handled here since
+		// onFocus excludes blocks involved in a multiselection, as
+		// focus can be incurred by starting a multiselection (focus
+		// moved to first block's multi-controls).
+		} else if ( isPartOfMultiSelection ) {
+			onSelect();
 		}
 	};
 
-	const onMouseUp = () => {
-		isPointerDown.current = false;
-	};
-
-	const onMouseLeave = () => {
-		if ( isPointerDown.current ) {
+	const onMouseLeave = ( event ) => {
+		// The primary button must be pressed to initiate selection.
+		// See https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons
+		if ( event.buttons === 1 ) {
```

Comment:
Looks like it might have some issues in Firefox in Mac and won't work in chrome Android https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons ?

Reply:
Oh no :)

Reply:
Ok, added a fallback. :)